integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -287,6 +291,172 @@ fn meta_tx_fn_call() {
     assert_eq!(fn_call_logs, &vec!["hello".to_owned()]);
 }
 
+/// Call a function in a meta tx where the user only has access through a
+/// function call access key.
+#[test]
+fn meta_tx_fn_call_access_key() {
+    let sender = bob_account();
+    let relayer = alice_account();
+    let receiver = carol_account();
+
+    let method_name = "log_something".to_owned();
+    let method_name_len = method_name.len() as u64;
+    let initial_allowance = NEAR_BASE;
+    let signer = create_user_test_signer(&sender);
+    let public_key = signer.public_key();
+    let access_key = AccessKey {
+        nonce: 0,
+        permission: AccessKeyPermission::FunctionCall(FunctionCallPermission {
+            allowance: Some(initial_allowance),
+            receiver_id: receiver.to_string(),
+            method_names: vec![method_name.clone()],
+        }),
+    };
+
+    let mut genesis = Genesis::test(vec![relayer.clone(), receiver.clone()], 3);
+    add_test_contract(&mut genesis, &receiver);
+    add_account_with_access_key(&mut genesis, &sender, NEAR_BASE, public_key.clone(), access_key);
+    let node = RuntimeNode::new_from_genesis(&relayer, genesis);
+
+    // Check previous allowance is set as expected
+    let key =
+        node.user().get_access_key(&sender, &public_key).expect("failed looking up fn access key");
+    let AccessKeyPermissionView::FunctionCall { allowance, ..} = key.permission else {
+        panic!("should be function access key")
+    };
+    assert_eq!(allowance.unwrap(), initial_allowance);
+
+    let actions = vec![Action::FunctionCall(FunctionCallAction {
+        method_name,
+        args: vec![],
+        gas: 30_000_000_000_000,
+        deposit: 0,
+    })];
+    let outcome = check_meta_tx_fn_call(
+        &node,
+        actions,
+        method_name_len,
+        0,
+        sender.clone(),
+        relayer,
+        receiver,
+    );
+
+    // Check that the function call was executed as expected
+    let fn_call_logs = &outcome.receipts_outcome[1].outcome.logs;
+    assert_eq!(fn_call_logs, &vec!["hello".to_owned()]);
+
+    // Check allowance was not updated
+    let key =
+        node.user().get_access_key(&sender, &public_key).expect("failed looking up fn access key");
+    let AccessKeyPermissionView::FunctionCall { allowance, ..} = key.permission else {
+        panic!("should be function access key")
+    };
+    assert_eq!(
+        allowance.unwrap(),
+        initial_allowance,
+        "allowance should not change, we used the relayer's fund not the sender's"
+    );
+}
+
+/// Call a function in a meta tx where the user only has access through a
+/// function call access that has zero allowance left.
```

Comment:
The comment does not match the implementation (`initial_allowance = 1`.)

Reply:
should be fixed now